src/java.base/share/classes/java/nio/X-Buffer.java.template

```diff
@@ -1666,16 +1666,11 @@ public abstract class $Type$Buffer
      * @return  A summary string
      */
     public String toString() {
-        StringBuffer sb = new StringBuffer();
-        sb.append(getClass().getName());
-        sb.append("[pos=");
-        sb.append(position());
-        sb.append(" lim=");
-        sb.append(limit());
-        sb.append(" cap=");
-        sb.append(capacity());
-        sb.append("]");
-        return sb.toString();
+        return getClass().getName()
+                 + "[pos=" + position()
+                 + " lim=" + limit()
+                 + " cap=" + capacity()
+                 + "]";
```

Comment:
This looks okay. Is this a drive-by observation or is something really stressing Buffer::toString?

Reply:
Viktor Klang made a drive-by observation here: https://twitter.com/viktorklang/status/1407688764567298052. I highly doubt it matters anywhere performance-wise, so I did no benchmarks.